engine/table/src/main/java/io/deephaven/engine/util/PythonDeephavenSession.java

```diff
@@ -82,12 +73,20 @@ public PythonDeephavenSession(ObjectTypeLookup objectTypeLookup, boolean runInit
      * @param runInitScripts if init scripts should be executed
      * @param isDefaultScriptSession true if this is in the default context of a worker jvm
      * @throws IOException if an IO error occurs running initialization scripts
+     * @throws InterruptedException if the current thread is interrupted while starting JPY
+     * @throws TimeoutException if jpy times out while obtaining configuration details
      */
     public PythonDeephavenSession(
             ObjectTypeLookup objectTypeLookup, @Nullable final Listener listener, boolean runInitScripts,
             boolean isDefaultScriptSession)
-            throws IOException {
+            throws IOException, InterruptedException, TimeoutException {
         super(objectTypeLookup, listener, isDefaultScriptSession);
+
+        //TODO in the py init code that would call in here, we should read the jpy system properties
```

Comment:
At the very least I need to remove this comment (doesnt belong in this branch), but this JpyInit.init(log) is probably the main objectionable thing here.

Reply:
Yeah, previously it was done in `initializeGlobalScriptSession`; DeephavenApiServer.run may be the "best" place for now?